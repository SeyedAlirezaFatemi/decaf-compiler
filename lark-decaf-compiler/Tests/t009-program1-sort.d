
void sort(int[] items) {

    /* implementation of bubble sort */
    int i;
    int j;

    int n;
    n = items.length();

    for (i = 0; i < n-1; i = i + 1)
        for (j = 0; j < n - i - 1; j = j + 1)
            if (items[j] > items[j + 1]) {
                int t;
                t = items[j];
                items[j] = items[j + 1];
                items[j + 1] = t;
            }
}

int main() {
    int i;
    int j;
    int[] rawitems;
    int[] items;

    Print("Please enter the numbers (max count: 100, enter -1 to end sooner): ");

    rawitems = NewArray(100, int);
    for (i = 0; i < 100; i = i + 1) {
        int x;
        x = ReadInteger();
        if (x == -1) break;

        rawitems[i] = x;
    }

    items = NewArray(i, int);

    // copy to a more convenient location
    for (j = 0; j < i; j = j + 1) {
        items[j] = rawitems[j];
    }

    sort(items);

    Print("After sort: ");

    for (i = 0; i < items.length(); i = i + 1) {
        Print(items[i]);
    }
}
