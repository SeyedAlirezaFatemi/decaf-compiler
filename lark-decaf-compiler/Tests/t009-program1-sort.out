Please enter the numbers (max count: 100, enter -1 to end sooner): 
After sort: 
0
1
2
3
3
5
5
8
8
8
9
10
10
11
11
12
13
14
14
14
14
17
17
17
18
18
19
19
19
19
21
22
24
25
28
28
28
28
29
29
29
31
37
41
42
48
51
53
53
53
54
55
56
57
57
59
60
61
62
64
65
68
70
74
74
74
77
77
77
79
81
82
82
82
83
85
86
87
88
89
91
94
95
97
97
97
97
98
98
99
